processing/src/main/java/org/apache/druid/segment/column/TypeStrategy.java

```diff
@@ -170,4 +172,30 @@ default T fromBytes(byte[] value)
   {
     throw new IllegalStateException("Not supported");
   }
+
+  /**
+   * Whether the type is groupable or not. This is always true for all the primitive types, arrays, and nested arrays
+   * therefore the SQL and the native layer might ignore this flag for those types. For complex types, this flag can be
+   * true or false, depending on whether the semantics and implementation of the type naturally leads to groupability
+   * or not. For example, it makes sense for JSON columns to be groupable, however there is little sense in grouping
+   * sketches (before finalizing).
+   *
+   * If a type is groupable, it MUST implement the {@link #hashCode} and {@link #equals} correctly
+   */
+  default boolean groupable()
+  {
+    return false;
+  }
+
+  @Override
+  default int hashCode(T o)
```

Comment:
Lets add java doc mentioning that most implementation have groupable=true while overriding this method.